pandas/tests/plotting/test_groupby.py

```diff
@@ -11,6 +11,45 @@
 
 from pandas.tests.plotting.common import TestPlotBase
 
+from itertools import count
+import pandas as pd
+import pytest
+
+
+@pytest.mark.parametrize(
+    'plotting_method',
+    ['line', 'bar', 'barh', 'box',
+     'area', 'pie', 'scatter', 'hexbin',
+     pytest.param('density', marks=td.skip_if_no_scipy),
+     pytest.param('kde', marks=td.skip_if_no_scipy)
+     ])
+@td.skip_if_no_mpl
+def test_no_double_plot_for_first_group(plotting_method):
+    class extGroupByPlot(pd.core.groupby.groupby.GroupByPlot):
```

Comment:
I think this test can be simplified.

How about

```python
df = pd.DataFrame({'cat': [1, 1, 2, 2], 'x': [1, 3, 5, 7], 'y': [2, 4, 6, 8]})
df.groupby('cat').plot.scatter(x='x', y='y')
assert plt.get_fignums() == [1, 2]
```

or something like that.

Reply:
You'd need to replace `.plot.scatter` with your `getattr(gr, plotting_method)(x='x', y='y')`, but the basic idea is to test the output, the number of produced figures, is the number of groups.

Reply:
That looks much better. I didn't know about `get_fignums`.

Reply:
Do you by any chance know how to reset this? It works for the first plot method, but then keep accumulating as `pytest.mark.parametrize` is looped through.

Reply:
Run `plt.close()` twice?

Reply:
Each test should close all its state. `plt.close('all')` should do it.